includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php

```diff
@@ -21,47 +20,47 @@ class WCS_Meta_Box_Related_Orders {
 
 	/**
 	 * Output the metabox
+	 * @param  WP_Post|WC_Order $post_or_order_object The post object or order object currently being edited.
 	 */
-	public static function output( $post ) {
-
-		if ( wcs_is_subscription( $post->ID ) ) {
-			$subscription = wcs_get_subscription( $post->ID );
-			$order = ( false == $subscription->get_parent_id() ) ? $subscription : $subscription->get_parent();
-		} else {
-			$order = wc_get_order( $post->ID );
-		}
+	public static function output( $post_or_order_object ) {
+		$order = ( $post_or_order_object instanceof WP_Post ) ? wc_get_order( $post_or_order_object->ID ) : $post_or_order_object;
 
 		add_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', __CLASS__ . '::output_rows', 10 );
 
-		include_once( dirname( __FILE__ ) . '/views/html-related-orders-table.php' );
+		include_once dirname( __FILE__ ) . '/views/html-related-orders-table.php';
 
-		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order, $post );
+		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order );
```

Comment:
We won't be able to remove the second here as third-parties hooking onto it may be using it and it would cause fatals. Unfortunately, we will need to deprecate the whole filter and continue passing a `post` object as long as we can (while we're still using CPT).

Reply:
Deprecated in 25c31be.